### DISCUSSION TITLE
[Laravel 11] Can't Update Database Even though Query Runs Without Error

### DISCUSSION THREAD

**Discussion by QLesson** [community/community#141501] [Programming Help]
### Body

Hello, I am facing a problem with Laravel 11. When I try to update data to the database, no error appears, but the data in the database does not change at all.

The steps I have tried:
1. Making sure the form uses the `POST` or `PUT` method.
2. Checking the `User` model is using `$fillable` or `$guarded` correctly.
3. Runs the query in `tinker`, the result is correct and appropriate.
4. Ensures the data is sent correctly using `dd($request->all())`.
5. Making sure the connection to the database is working properly.

Here's the code snippet I used:

```php
// Route
Route::put('/profile/update', [ProfileController::class, 'update'])->name('profile.update');

// Controller
public function update(Request $request, User $user)
{
    $validatedData = $request->validate([
        'name' => 'required|max:50',
        'email' => 'required|email',
    ]);

    $user->update($validatedData); // It should update the data, but there are no changes in the DB

    return redirect()->back()->with('success', 'Profile updated');
}
```

I expected the data to be updated in the database after the form was submitted. However, although a success message appears, the data in the database does not change.

The version I am using:

Laravel: 11
PHP: 8.3
MySQL: 8.0
OS: Windows 11

Can anyone help me understand what might be wrong or what steps I'm missing? Thank you very much!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by aldiipratama** [comment]
Make sure that in the User model, you have defined the $fillable or $guarded property so that Laravel knows what fields can be filled via the update() method.

Example:
```php
User class extends Authenticatable
{
    protected $contents = ['name', 'email', 'profile_image']; // Columns that can be updated
}
```

If you are using $guarded, make sure you are not blocking the column you want to update:
```php
protected $guarded = ['id']; // Other columns can be updated except 'id'
```

Make sure the data you validate matches the columns in the database and is correct.

Example:
```php
$validatedData = $request->validate([
    'name' => 'required|max:50',
    'email' => 'required|email',
    'profile_image' => 'nullable|image|max:5120' // Add if any files are uploaded
]);
```

To check if the update query was actually executed, you can use Laravel's Query Log by enabling query logging in the AppServiceProvider.

Add this to boot() in AppServiceProvider:
```php
\DB::listen(function ($query) {
    \Log::info($query->sql);
});
```

This will write all executed queries to a log file, so you can see if the UPDATE query was executed or not.

Make sure the form you are using uses the correct PUT or POST method, according to the route you are using.

Example form:
```php
<form action = “{{ route(‘profile.update’, $user->username)}}” method = “POST” enctype = “multipart/form-data”>
    @csrf
    @method('PUT')
    <!-- Input field -->
</form>
```

Make sure you include @csrf and @method('PUT') if using the PUT route.

**Comment by ebndev** [comment]
Hi @QLesson & @aldiipratama, thanks for being a part of the GitHub Community!

While we appreciate you wanting to contribute to Discussions, please note: We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Note that GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this, and other related posts. 

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
